internal/characters/thoma/attack.go

```diff
@@ -9,27 +9,42 @@ import (
 	"github.com/genshinsim/gcsim/pkg/core/combat"
 )
 
-var attackFrames [][]int
-var attackHitmarks = [][]int{{11}, {38}, {27, 40}, {25}}
-var attackHitlagHaltFrame = [][]float64{{0.06}, {0.09}, {0, 0}, {0}}
-var attackDefHalt = [][]bool{{true}, {true}, {false, false}, {false}}
+var (
+	attackFrames          [][]int
+	attackHitmarks        = [][]int{{13}, {18}, {10, 13}, {13}}
+	attackHitlagHaltFrame = [][]float64{{0.06}, {0.09}, {0, 0}, {0}}
+	attackDefHalt         = [][]bool{{true}, {true}, {false, false}, {false}}
+)
 
 const normalHitNum = 4
 
 func init() {
 	attackFrames = make([][]int, normalHitNum)
 
-	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 11)
-	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 38)
-	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][1], 40)
-	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][0], 25)
-	attackFrames[3][action.ActionCharge] = 500 //TODO: this action is illegal; need better way to handle it
+	// N1 -> x
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 21)
+	attackFrames[0][action.ActionAttack] = 20
+	attackFrames[0][action.ActionCharge] = 21
+
+	// N2 -> x
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 27)
+	attackFrames[1][action.ActionAttack] = 27
+	attackFrames[1][action.ActionCharge] = 25
+
+	// N3 -> x
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][1], 32)
+	attackFrames[2][action.ActionAttack] = 31
+	attackFrames[2][action.ActionCharge] = 32
+
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][0], 58)
+	attackFrames[3][action.ActionAttack] = 58
+	attackFrames[3][action.ActionCharge] = 500 // TODO: this action is illegal; need better way to handle it
 }
 
 // Normal attack damage queue generator
 // relatively standard with no major differences versus other characters
 func (c *char) Attack(p map[string]int) action.ActionInfo {
-
+	lastMultiHit := 0
```

Comment:
you don't need to do this if you use {10, 23} instead of {10, 13} for N3 and go back to using attackHitmarks[c.NormalCounter][i] as delay for the task